rebasebot/bot.py

```diff
@@ -137,32 +138,66 @@ def _add_to_rebase(commit_message, source_repo, tag_policy):
     return tag_policy == "soft"
 
 
-def _do_rebase(gitwd, source, dest, source_repo, tag_policy):
+def _do_rebase(gitwd, source, dest, source_repo, tag_policy, allow_bot_squash, bot_emails):
     logging.info("Performing rebase")
 
     merge_base = gitwd.git.merge_base(f"source/{source.branch}", f"dest/{dest.branch}")
     logging.info("Rebasing from merge base: %s", merge_base)
 
     # Find the list of commits between the merge base and the destination head
     # This should be the list of commits we are carrying on top of the UPSTREAM
-    commits = gitwd.git.log("--reverse", "--pretty=format:%H - %s", "--no-merges",
+    commits = gitwd.git.log("--reverse", "--pretty=format:%H - %s - %aE", "--no-merges",
                             "--ancestry-path", f"{merge_base}..dest/{dest.branch}")
     logging.info("Picking commits: \n%s", commits)
 
+    commits_to_squash = defaultdict(list)
+
     for commit in commits.splitlines():
         # Commit contains the message for logging purposes,
         # trim on the first space to get just the commit sha
-        sha, commit_message = commit.split(" - ")
+        sha, commit_message, committer_email = commit.split(" - ")
 
         if not _add_to_rebase(commit_message, source_repo, tag_policy):
             continue
 
         try:
-            gitwd.git.cherry_pick(f"{sha}", "-Xtheirs")
+            if allow_bot_squash:
+                # There is sometimes a prefix with number and a following + sign
+                # We have to get rid of that part to make sure to get
+                # only the email of the bot
+                is_plus_in_email = committer_email.find("+")
```

Comment:
lines 168 - 173 might be replaced with one line, I think
`email = comitter_email.split("+")[-1]`

Reply:
Done.